SignalMessaging/attachments/MediaMessageView.swift

```diff
@@ -277,6 +279,58 @@ public class MediaMessageView: UIView, OWSAudioAttachmentPlayerDelegate {
         }
     }
 
+    private func createTextPreview() {
+
+        let data = attachment.data
+        guard let messageText = String(data: data, encoding: String.Encoding.utf8) else {
+            createGenericPreview()
+            return
+        }
+
+        let messageBubbleView = UIImageView()
+        messageBubbleView.layoutMargins = .zero
+        let bubbleImageData =
+            OWSMessagesBubbleImageFactory.shared.outgoing
+        messageBubbleView.image = bubbleImageData.messageBubbleImage
+
+        let textColor = UIColor.white
+
+        let messageTextView = UITextView()
+        messageTextView.font = UIFont.ows_dynamicTypeBody()
+                messageTextView.backgroundColor = UIColor.clear
+        messageTextView.isOpaque = false
+        messageTextView.isEditable = false
+        messageTextView.isSelectable = false
+        messageTextView.textContainerInset = UIEdgeInsets.zero
+        messageTextView.contentInset = UIEdgeInsets.zero
+        messageTextView.isScrollEnabled = false
+        messageTextView.showsHorizontalScrollIndicator = false
+        messageTextView.showsVerticalScrollIndicator = false
+        messageTextView.isUserInteractionEnabled = false
+        messageTextView.textColor = textColor
+        messageTextView.linkTextAttributes = [NSForegroundColorAttributeName : textColor,
+                                              NSUnderlineStyleAttributeName : [NSUnderlineStyle.styleSingle,
+                                                                               NSUnderlineStyle.patternSolid]
+                                              ]
+        messageTextView.dataDetectorTypes = [.link, .address, .calendarEvent]
+        messageTextView.text = messageText
+
+        messageBubbleView.layoutMargins = .zero
+        self.layoutMargins = .zero
+
+        self.addSubview(messageBubbleView)
+        messageBubbleView.autoVCenterInSuperview()
+        messageBubbleView.autoHCenterInSuperview()
+        messageBubbleView.autoPinEdge(toSuperviewEdge: .leading, withInset: 25, relation: .greaterThanOrEqual)
+        messageBubbleView.autoPinEdge(toSuperviewEdge: .trailing, withInset: 25, relation: .greaterThanOrEqual)
+
+        messageBubbleView.addSubview(messageTextView)
+        messageTextView.autoPinTopToSuperview(withMargin:10)
+        messageTextView.autoPinBottomToSuperview(withMargin:10)
+        messageTextView.autoPinLeadingToSuperview(withMargin:10)
+        messageTextView.autoPinTrailingToSuperview(withMargin:15)
```

Comment:
As discussed, I'd like to rework bubble appearance soon.  When we do that, I'll pull out these constants to somewhere in `SignalMessaging`.